nutanix/data_source_nutanix_foundation_central_imaged_node_details.go

```diff
@@ -0,0 +1,285 @@
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cvm_gateway": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cvm_ip": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cvm_ipv6": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cvm_netmask": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cvm_up": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"cvm_uuid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cvm_vlan_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"foundation_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hardware_attributes": {
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
+			"hypervisor_gateway": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hypervisor_hostname": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hypervisor_ip": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hypervisor_netmask": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hypervisor_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hypervisor_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"imaged_cluster_uuid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ipmi_gateway": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ipmi_ip": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ipmi_netmask": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ipv6_interface": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"model": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"node_position": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"node_serial": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"node_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"node_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"object_version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+			"latest_hb_ts_list": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceFCImagedNodeDetailsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).FC
+	req := fc.ImagedNodeDetailsInput{}
+
+	nodeUUID, ok := d.GetOk("imaged_node_uuid")
+	if !ok {
+		return diag.Errorf("please provide the imaged_node_uuid")
+	}
+	req.ImagedNodeUUID = nodeUUID.(string)
+
+	res, err := conn.GetImagedNode(ctx, req.ImagedNodeUUID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("aos_version", (res.AosVersion)); err != nil {
```

Comment:
```suggestion
	if err := d.Set("aos_version", res.AosVersion); err != nil {
```

Make changes everywhere.

Reply:
will change in next PRs.